### ISSUE TITLE
Venue is changed if the scheduler doesn't access the database

### ISSUE THREAD

**Issue by romw** [BOINC/boinc-ng#556]
**Reported by knreed on 11 Dec 38198675 08:53 UTC**
In the event that the scheduler does not access the database, for example if there is no work or if the scheduler cannot access the database, then the scheduler reply will look something like:
```xml
<scheduler_reply>
<scheduler_version>601</scheduler_version>
<master_url>http://worldcommunitygriddev.com/</master_url>
<request_delay>3600.000000</request_delay>
<message priority="low">Server can't open database </message>
<project_name>World Community Grid (Dev)</project_name>
</scheduler_reply>
```

The field host_venue is initialized to the empty string during parsing of the response and since it is never populated (since the tag is not present), the client then uses the empty string as the value from the server.  If the client had previously had a different venue set such as 'home' the client would then decide that the venue had changed and will report:
```
3/12/2008 8:19:24 PM|World Community Grid (Dev)|Sending scheduler request: Requested by user.  Requesting 0 seconds of work, reporting 0 completed tasks
3/12/2008 8:19:29 PM|World Community Grid (Dev)|Scheduler request succeeded: got 0 new tasks
3/12/2008 8:19:29 PM|World Community Grid (Dev)|Message from server: Server can't open database
3/12/2008 8:19:29 PM|World Community Grid (Dev)|New host venue: 
```

This will then cause the client to use the default venue rather then its assigned venue which could cause behavior on the client to change that was not desired by the member.

**Comment by romw** [comment]
**Commented by Ageless on 13 Oct 38198747 19:06 UTC**
5.10 has ended with 5.10.45, any new changes will be set to the next milestone. Setting milestone to 6.2 for the new client version.

**Comment by romw** [comment]
**Commented by knreed on 30 Apr 38200807 14:13 UTC**
A code update to the scheduler has been sent to David that should close this issue.

**Comment by romw** [comment]
**Commented by davea on 7 Nov 38201201 19:33 UTC**
(In [14908]) - Validator framework: remove some consts, and other changes,
    to allow validator to assign different credit
    to different instances of a job
- Scheduler: if can't open DB, return <project_is_down/>
    (fixes #578)
- clean up logic of modify_claimed_credit
- feeder: for -priority_order_create_time, use workunitid
    rather than create time (faster for the DB)
from Kevin Reed
